ivy_tests/test_ivy/helpers/hypothesis_helpers/dtype_helpers.py

```diff
@@ -50,9 +50,13 @@ def _get_type_dict(framework: str, kind: str, is_frontend_test=False):
 def _get_type_dict_helper(framework, kind, is_frontend_test):
     if is_frontend_test:
         framework_module = get_frontend_config(framework).supported_dtypes
+        return _retrieve_requested_dtypes(framework_module, kind)
     else:
-        framework_module = ivy
+        with WithBackendContext(framework) as ivy_backend:
+            return _retrieve_requested_dtypes(ivy_backend, kind)
 
+
+def _retrieve_requested_dtypes(framework_module, kind):
```

Comment:
we don't need to write a new function for this, you can simply assign the changes to framework_module, the changes made within context will persists.